tiledb/sm/c_api/tiledb_experimental.h

```diff
@@ -171,6 +171,454 @@ TILEDB_EXPORT int32_t tiledb_array_evolve(
+ * tiledb_subarray_partitioner_t *subarray_partitioner;
+ * tiledb_subarray_partitioner_set_layout(ctx, layout, subarray_partitioner);
+ * @endcode
+ *
+ * @param layout The layout to set into the subarray partioner object's
+ * subarray.
+ *    - `TILEDB_COL_MAJOR`:
+ *      This means column-major order with respect to the subarray.
+ *    - `TILEDB_ROW_MAJOR`:
+ *      This means row-major order with respect to the subarray.
+ *    - `TILEDB_GLOBAL_ORDER`:
+ *      This means that cells are stored or retrieved in the array global
+ *      cell order.
+ *    - `TILEDB_UNORDERED`:
+ *      This is applicable only to reads and writes for sparse arrays, or for
+ *      sparse writes to dense arrays. For writes, it specifies that the cells
+ *      are unordered and, hence, TileDB must sort the cells in the global cell
+ *      order prior to writing. For reads, TileDB will return the cells without
+ *      any particular order, which will often lead to better performance.
+ * * @return `TILEDB_OK` for success or `TILEDB_ERR` for error.
+ */
+TILEDB_EXPORT int32_t tiledb_subarray_partitioner_set_layout(
+    tiledb_ctx_t* ctx,
+    tiledb_layout_t layout,
+    tiledb_subarray_partitioner_t* partitioner);
+
+/**
+ * Compute the complete series of partitions/subarrays leaving them stored
+ * internally in the object.
+ *
+ * These partitions can then be retrieved via
+ * tiledb_subarray_partitioner_get_partition().
+ *
+ * **Example:**
+ *
+ * @code{.c}
+ * tiledb_subarray_partitioner_t *subarray_partitioner;
+ * tiledb_subarray_partitioner_compute(ctx, subarray_partitioner);
+ * @endcode
+ *
+ * @param The partitioner for which to compute the complete series of (sub)
+ * partitions
+ * @return `TILEDB_OK` for success or `TILEDB_ERR` for error.
+ */
+TILEDB_EXPORT int32_t tiledb_subarray_partitioner_compute(
+    tiledb_ctx_t* ctx, tiledb_subarray_partitioner_t* partitioner);
+
+/**
+ * Gets number of computed partitions available to be retrieved via
+ * tiledb_subarray_partitioner_get_partition().
+ *
+ * **Example:**
+ *
+ * @code{.c}
+ * tiledb_subarray_partitioner_t *subarray_partitioner;
+ * tiledb_subarray_partitioner_compute(ctx, subarray_partitioner);
+ * @endcode
+ *
+ * @param layout The layout to set into the subarray partioner object's
+ * subarray.
+ * @return `TILEDB_OK` for success or `TILEDB_ERR` for error.
+ */
+TILEDB_EXPORT int32_t tiledb_subarray_partitioner_get_partition_num(
+    tiledb_ctx_t* ctx,
+    uint64_t* num,
+    tiledb_subarray_partitioner_t* partitioner);
+
+/**
+ * Retrieve the part-id(th) partition subarray from the partitioner's computed
+ * subarray partitions.
+ *
+ * **Example:**
+ *
+ * @code{.c}
+ * uint64_t part_id; //# of partition to retrieve, must be less than # computed
+ * partitions tiledb_subarray_t *retrieved_subarray;
+ * tiledb_subarray_partitioner_t *subarray_partitioner;
+ * tiledb_subarray_partitioner_get_partition(ctx, subarray_partitioner,
+ * partition_id, retrieved_subarray);
```

Comment:
named part_id above.

Reply:
used partition_id above and also changed within ..._get_partition() prototype and definition.